wcomponents-core/src/main/java/com/github/bordertech/wcomponents/ComponentModel.java

```diff
@@ -537,25 +537,95 @@ protected void setAccessibleText(final String text, final Serializable... args)
 		this.accessibleText = I18nUtilities.asMessage(text, args);
 	}
 
+
 	/**
-	 * @return Returns the HTML class.
+	 * @return the HTML class list as a single space separated String.
 	 */
 	protected Serializable getHtmlClass() {
-		return htmlClass;
+		if (this.htmlClassList == null || this.htmlClassList.isEmpty()) {
+			return null;
+		}
+		int i = 0;
+		StringBuilder result = new StringBuilder();
+
+		for (String c : this.htmlClassList) {
+			if (i++ > 0) {
+				result.append(" ");
+			}
+			result.append(c);
+		}
+		return result.toString();
 	}
 
 	/**
 	 * @param text The HTML class name text to set.
 	 */
 	protected void setHtmlClass(final String text) {
-		this.htmlClass = text;
+		if (text == null) {
+			this.htmlClassList = null;
+		} else {
+			this.htmlClassList = new HashSet<>();
+			this.htmlClassList.add(text);
+		}
 	}
 
 	/**
 	 * @param className The HTML class name to set.
 	 */
 	protected void setHtmlClass(final HtmlClassProperties className) {
-		this.htmlClass = className.toString();
+		if (null == className) {
+			this.htmlClassList = null;
+		} else {
+			setHtmlClass(className.toString());
+		}
+	}
+
+	/**
+	 * @param text The HTML class name text to set.
+	 */
+	protected void addHtmlClass(final String text) {
+		if (!Util.empty(text)) {
+			if (this.htmlClassList == null) {
+				this.htmlClassList = new HashSet<>();
+			}
+			this.htmlClassList.add(text);
+		}
+	}
+
+	/**
+	 * @param className The HTML class name to set.
+	 */
+	protected void addHtmlClass(final HtmlClassProperties className) {
+		if (null != className) {
+			addHtmlClass(className.toString());
+		}
+	}
+
+	/**
+	 * @return the set of HTML classes to add to the current component.
+	 */
+	protected HashSet getHtmlClassList() {
+		return this.htmlClassList;
+	}
+
+	/**
+	 * Remove a value from the HTML class name list.
+	 * @param className the value to remove
+	 */
+	protected void removeHtmlClass(final String className) {
+		if (!(this.htmlClassList == null || Util.empty(className))) {

```

Comment:
I would tend to code this without the wrapping negative....`this.htmlClassList != null && !Util.empty(className)`...but that is just me.